charts/incubator/nextcloud/docs/migration-guide.md

```diff
@@ -0,0 +1,155 @@
+# Nextcloud Migration Guide
+
+This guide will walk you through the migration process for Nextcloud. This guide utilize HostPath or NFS for UserData.
+
+:::danger TWO FACTOR AUTHENTICATION
+
+If you are using two factor authentication on your Nextcloud accounts disable it before starting this guide. While it can be done via the CLI later it is much easier to disable it before hand.
+
+:::
+
+## Backup Nextcloud Database
+
+If you have not already done so install PGAdmin and the [tcdbinfo.sh script](https://truecharts.org/manual/SCALE/guides/sql-export#how-to-list-database-login-info-for-truecharts-apps)
+
+1. Run the tcdbinfo.sh script and take note of the connection info
+
+```bash
+bash tcdbinfo.sh
+```
+
+![tcdbinfo-output](./img/tcdbinfo-output.png)
+
+2. Add Nextcloud to PGAdmin using the connection info from previous step.
+
+![pgadmin-connection1](./img/pgadmin-connection1.png)
+![pgadmin-connection2](./img/pgadmin-connection2.png)
+
+3. Backup Nextcloud database.
+
+![pgadmin-backup](./img/pgadmin-backup.png)
+
+## Delete Nextcloud
+
+:::danger Userdata Location
+
+Do not remove your Nextcloud installation until you have confirmed you use HostPath/NFS for user data or have copied UserData using utilizing [Heavyscript](https://github.com/Heavybullets8/heavy_script)
+
+:::
+
+If you are using HostPath of NFS for UserData then remove your current Nextcloud installation. If you are using PVC for UserData then you will first need to mount your Nextcloud PVC and copy your UserData to a new dataset.
+
+## Remove Old Nextcloud AppData
+
+This step is not required but to have a clean installation remove all UserData folder and files that are not user folders.
+
+```bash
+ls -la
+```
+
+![userdata-old](./img/userdata-old.png)
+
+```bash
+rm -rf <folder>
+```
+```bash
+rm .ocdata .htaccess index.html nextcloud.log
+```
```

Comment:
```suggestion

:::warning
Next steps are destructive without a way to undo.
Make sure you have proper backups to avoid data loss.

**You have been warned**

```bash
rm -rf <folder>
```
```bash
rm .ocdata .htaccess index.html nextcloud.log
```
:::

```